scrapy_autoextract/providers.py

```diff
@@ -0,0 +1,98 @@
+from typing import ClassVar, Optional, Type
+
+from autoextract.aio import request_raw
+from autoextract.request import Request as AutoExtractRequest
+from autoextract_poet.page_inputs import (
+    AutoExtractArticleData,
+    AutoExtractProductData,
+)
+from scrapy import Request
+from scrapy.settings import Settings
+from scrapy.statscollectors import StatsCollector
+from scrapy_poet.page_input_providers import (
+    PageObjectInputProvider,
+    register,
+)
+
+
+class _Provider(PageObjectInputProvider):
+    """An interface that describes a generic AutoExtract Provider.
+
+    It should not be used publicly as it serves the purpose of being a base
+    class for more specific providers such as Article and Product providers.
+    """
+
+    provided_class: ClassVar[Optional[Type]]
+
+    def __init__(
+            self,
+            request: Request,
+            settings: Settings,
+            stats: StatsCollector,
+    ):
+        """Initialize provider storing its dependencies as attributes."""
+        self.request = request
+        self.stats = stats
+        self.settings = settings
+
+    async def __call__(self):
+        """Make an AutoExtract request and build a Page Input of provided class
+        based on API response data.
+        """
+        page_type = self.get_page_type()
+        self.stats.inc_value(f"autoextract/{page_type}/total")
+
+        request = AutoExtractRequest(
+            url=self.request.url,
+            pageType=page_type,
+        )
+        api_key = self.settings.get("AUTOEXTRACT_USER")
+        endpoint = self.settings.get("AUTOEXTRACT_URL")
+        max_query_error_retries = self.settings.getint(
+            "AUTOEXTRACT_MAX_QUERY_ERROR_RETRIES", 3
+        )
+
+        try:
+            data = await request_raw(
+                [request],
+                api_key=api_key,
+                endpoint=endpoint,
+                max_query_error_retries=max_query_error_retries
+            )
+        except Exception:
+            self.stats.inc_value(f"autoextract/{page_type}/error")
+            raise
+
+        self.stats.inc_value(f"autoextract/{page_type}/success")
```

Comment:
We should check here for the content. If `error` in `data[0]` then raise `QueryLevelError`.  Only True successful requests should pass without exception.

Reply:
Nice observation, Iván! I've just refactored this code to raise an exception when the response data contains an error.